lms/djangoapps/certificates/api.py

```diff
@@ -308,8 +309,16 @@ def certificate_downloadable_status(student, course_key):
         'download_url': None,
         'uuid': None,
     }
-    may_view_certificate = CourseOverview.get_from_id(course_key).may_certify()
 
+    course_overview = CourseOverview.get_from_id(course_key)
+    if (
+        not certificates_viewable_for_course(course_overview) and
```

Comment:
Does this do any db work? Might want it last in the boolean sequence if so.

Reply:
It does not. Just checks a bunch of fields on the course object passed in. See https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/certificates/api.py#L31